### PR TITLE
injiweb-1607-sec-backward-navigation added logout confirmation modal

### PR DESCRIPTION
**Pull request by cyber-titan**
## Summary by CodeRabbit

* **New Features**
  * Adds a logout confirmation modal in the user area that appears on back navigation with "Logout" and "Stay" actions and integrated logout handling.

* **Localization**
  * Adds LogoutConfirmation translations for Arabic, English, French, Hindi, Kannada, Portuguese, and Tamil; fixes Portuguese logout labels.

* **Style**
  * Introduces centralized styles for the logout confirmation modal.

* **Tests**
  * Expands tests for the modal, logout/navigation guard, interactions, accessibility, and event cleanup.

* **Bug Fixes**
  * Uses history-safe redirect for Google login to avoid adding the IdP page to browser history.

* **Refactor**
  * Clickable component props updated to allow standard HTML attributes.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: inji-web/src/components/User/Layout.tsx [hunk 1]
```diff
@@ -47,6 +65,75 @@ export const Layout: React.FC = () => {
         return () => window.removeEventListener('resize', updateHeights);
     }, []);
 
+    useEffect(() => {
+        try {
+            const currentState: any = window.history.state || {};
+            // install only once per session (per tab) when inside /user/* layout
+            if (!currentState?.injiSentinelInstalled) {
```
Comment (jackjain) [inline inji-web/src/components/User/Layout.tsx]: Can we avoid adding a new state in window history ? Will the logic still work without it. Also please test across multiple browsers - Edge, Safari, Chrome and Firefox

File: inji-web/src/modals/LogoutConfirmationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+/**
+ * LogoutConfirmationModal component
+ *
+ * Features:
+ * - Displays a confirmation modal when user tries to navigate back from protected routes
+ * - Shows an info icon, warning message about logout, and two action buttons
+ * - Handles session cleanup on logout confirmation
+ * - Allows user to stay on current page
+ */
+export interface LogoutConfirmationModalProps {
```
[the remark this answers is not in the document]
Comment (jackjain) [reply inji-web/src/modals/LogoutConfirmationModal.tsx]: Lets add a screenshot of the mobile view in the bug card

File: inji-web/src/modals/LogoutConfirmationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                            className={LogoutConfirmationModalStyles.stayButton}
+                        >
+                            {t('stayButton')}
+                        </button>
+                    </div>
+                </div>
+            </div>
+        </Clickable>,
+        document.body
+    );
```
[the remark this answers is not in the document]
Comment (cyber-titan) [reply inji-web/src/modals/LogoutConfirmationModal.tsx]: Added rest except escape key handler logic

File: inji-web/src/components/User/Layout.tsx [hunk 4]
```diff
@@ -47,6 +67,93 @@ export const Layout: React.FC = () => {
...
+            try {
+                isGuardInstalledInSession = window.sessionStorage.getItem(NAV_GUARD_SESSION_KEY) === 'true';
+            } catch (e) {
+                console.warn('[nav-guard] Unable to access sessionStorage', e);
+            }
+            
+            // Install guard only once per tab session to avoid growing history on each navigation.
+            if (!isGuardInstalledInSession) {
+                window.history.replaceState(
+                    { logoutConfirmationGuard: true, navigationGuardInstalled: true },
```
[the remark this answers is not in the document]
Comment (cyber-titan) [reply inji-web/src/components/User/Layout.tsx]: removed as it is not needed now

File: inji-web/src/components/User/Layout.tsx [hunk 5]
```diff
@@ -47,6 +67,93 @@ export const Layout: React.FC = () => {
...
+                }
+            }
+        } catch (error) {
+            // As a fallback, clear client state and redirect
+            window.sessionStorage.removeItem(NAV_GUARD_SESSION_KEY);
+            removeUser?.();
+            setShowLogoutModal(false);
+            window.location.replace(ROUTES.ROOT);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (cyber-titan) [reply inji-web/src/components/User/Layout.tsx]: **1. Cleanup first:** For logout our primary goal is to reliably clear client auth state and redirect, even if the backend call fails. Leaving the user in a half logged‑in state is a desired behavior, so the catch block intentionally performs the same cleanup and redirect.

**2. user_logout_error as success:** That error code is already used in the codebase to mean “session already invalid/terminated”, which is effectively a successful logout from the user’s perspective.

**3. Helper for duplication:** I’ll extract the shared cleanup (remove , close modal, redirect) into a small helper to make the behavior clearer and easier to maintain, without changing the existing logic.
